src/acl/Checklist.cc

```diff
@@ -60,6 +60,7 @@ ACLChecklist::markFinished(const Acl::Answer &finalAnswer, const char *reason)
     assert (!finished() && !asyncInProgress());
     finished_ = true;
     answer_ = finalAnswer;
+    answer_.lastCheckedName = lastCheckedName_;
     debugs(28, 3, this << " answer " << answer_ << " for " << reason);
```

Comment:
We thought of a new way to test PR changes. These new tests did not uncover PR bugs, but we have discovered that ACLChecklist is missing a couple of markFinished() calls. Adding those calls will improve reconfiguration support[^1], but the best solution is not to sprinkle ACLChecklist with more markFinished() calls. Instead, we should refactor a bit so that we will never forget to make those calls again. That non-trivial refactoring is outside this PR scope. Once this PR lands, we will follow up with a PR dedicated to that ACLChecklist refactoring.

[^1]: Reconfiguration support will improve by delivering the last evaluated ACL name to nonBlockingCheck() callbacks. Today, that name is missing in the official code because that code is missing markFinished() calls _and_ because of the various AclMatchedName-linked issues addressed in this PR. That name remains missing in PR code because those markFinished() calls remain missing in PR code. Again, we will address the underlying problem in a dedicated followup PR to keep this PR focused on Acl::Node::name/AclMatchedName upgrade/fixes.

Reply:
Posted this follow-up [PR](https://github.com/squid-cache/squid/pull/1809).